docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -1331,6 +1331,30 @@ public void register(RegistrationContext context) {
 
 <1> The argument is the bindings source class.
 
+=== `InstanceHandle.close()` Behavior
+
+Per the CDI specification, the `Instance.Handle.close()` method always delegates to `destroy()`.
+In ArC, this is only true in the <<strict_mode>>.
+
+In the default mode, the `close()` method only delegates to `destroy()` when the bean is `@Dependent` (or when the instance handle does not represent a CDI contextual object).
+When the instance handle represents a bean of any other scope, the `close()` method does nothing; the bean is left as is and will be destroyed whenever its context is destroyed.
+
+This is to make the following code behave as one would naively expect:
+
+[source,java]
+----
+Instance<T> instance = ...;
+for (Instance.Handle<T> handle : instance.handles()) {
```

Comment:
I would personally avoid the loop in the snippet and simplify to something like:
```
Instance<T> instance = ...;
try (Instance.Handle<T> handle : instance.getHandle()) {
   T t = handle.get();
   // ... use t...
}
```
But that's just a matter of taste...

Reply:
Good point, will do.

Reply:
Done.